core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsBasicCredentials.java

```diff
@@ -111,14 +128,59 @@ public boolean equals(Object o) {
         }
         AwsBasicCredentials that = (AwsBasicCredentials) o;
         return Objects.equals(accessKeyId, that.accessKeyId) &&
-               Objects.equals(secretAccessKey, that.secretAccessKey);
+               Objects.equals(secretAccessKey, that.secretAccessKey) &&
+               Objects.equals(credentialScope, that.credentialScope().orElse(null));
     }
 
     @Override
     public int hashCode() {
         int hashCode = 1;
         hashCode = 31 * hashCode + Objects.hashCode(accessKeyId());
         hashCode = 31 * hashCode + Objects.hashCode(secretAccessKey());
+        hashCode = 31 * hashCode + Objects.hashCode(credentialScope());
         return hashCode;
     }
+
+    public static final class Builder {
```

Comment:
Copyable Builder?

Reply:
keeping consistent with AwsSessionCredentials, do we want to make both of them Copyable Builder?

Reply:
Gotcha, can we use interface since this is public API?

Reply:
added Builder interface for AwsBasicCredentials and AwsSessionCredentials